docs/javascript-unit-tests.md

```diff
@@ -0,0 +1,712 @@
+> 2.  Write only enough production code to make the failing unit test pass.
+
+Then repeat that process until you have written all of the code you need.
+
+[Read this primer on test-driven development](https://medium.freecodecamp.org/test-driven-development-what-it-is-and-what-it-is-not-41fa6bca02a2?gi=3c8d8b476cc9)
+to learn more about how it differs from
+the typical approach to programming and unit tests.
+
+
+
+
+## Setting up tests
+
+
+### New test file from sample
+
+For this guide, we’ll use very basic sample code files
+to illustrate how to use the test framework in cfgov-refresh
+and how to test very common code patterns.
+
+Another common approach is to look for existing tests
+that are testing something similar to what you are writing now.
+Feel free to do so and substitute another test file for
+the `sample-spec.js` file referenced below.
+For links to existing tests,
+[see the “Running tests” section](#running-tests) on this page.
+
+Now, let’s begin! Let’s make a new unit test fail,
+then we will make it pass, following the principles of TDD.
+
+1. Create 2 new files: a test file, and a JavaScript file.
+   Their names should match, with the test file adding a `-spec` suffix.
+   For example, if your script is named `my-javascript.js`,
+   then your test is named `my-javascript-spec.js`.
+1. Copy the code from this
+   [`sample-spec.js` file](https://github.com/cfpb/cfgov-refresh/blob/fc9ecc002f933a432322bd15f54be5eb2cffa6e3/test/unit_tests/js/modules/sample-spec.js)
+   into your own test file.
+1. Copy and paste from
+   [`sample.js`](https://github.com/cfpb/cfgov-refresh/blob/fc9ecc002f933a432322bd15f54be5eb2cffa6e3/cfgov/unprocessed/js/modules/sample.js)
+   into your new JavaScript file.
+
+
+### Folder structure (where to put your JavaScript and tests)
+
+JavaScript unit test files belong in the
+[`test/unit_tests/`](https://github.com/cfpb/cfgov-refresh/tree/master/test/unit_tests)
+directory.
+
+The folder structure of the test files mirrors
+the structure of the project JavaScript in
+[`cfgov/unprocessed/js/`](https://github.com/cfpb/cfgov-refresh/tree/master/cfgov/unprocessed/js).
+
+So my `sample-spec.js` file is in `test/unit_tests/modules/`
+and my `sample.js` file is in `cfgov/unprocessed/js/modules/`.
+I put these files in the `modules` folder because
+they are not atomic components, they are just sample files.
+Your JavaScript might be an atomic element,
+in which case it should go in the corresponding folder.
+If you're not sure where your JavaScript should go,
+[read about atomic components in cfgov-refresh](../atomic-structure/).
+
+!!! note "Child apps"
+    If you’re working on something in a child app,
+    put it in `test/unit_test/appname/js/`.
+    Otherwise, if you’re working on something that
+    belongs to cfgov-refresh generally,
+    it should go in the corresponding folder under `test/unit_test/js/`.
+
+Now that you have your sample JS and test files in the right places,
```

Comment:
Trivial: JS used here, JavaScript used elsewhere. Probably should have `JavaScript (JS)` at the first occurrence in the doc.